src/rpc/mining.cpp

```diff
@@ -228,6 +297,98 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
     return generateBlocks(mempool, coinbase_script, nGenerate, nMaxTries);
 }
 
+static UniValue generateblock(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"generateblock",
+        "\nMine a block with a set of transactions immediately to a specified address or descriptor (before the RPC call returns)\n",
+        {
+            {"address/descriptor", RPCArg::Type::STR, RPCArg::Optional::NO, "The address or descriptor to send the newly generated bitcoin to."},
+            {"transactions", RPCArg::Type::ARR, RPCArg::Optional::NO, "An array of hex strings which are either txids or raw transactions.\n"
+                "Txids must reference transactions currently in the mempool.",
+                {
+                    {"rawtx/txid", RPCArg::Type::STR_HEX, RPCArg::Optional::OMITTED, ""},
+                },
+            }
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
```

Comment:
I think the string does not need to be wrapped in an object and can be returned as is.

Reply:
I was attempting to follow the guidelines from https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#rpc-interface-guidelines: 

- Try to make the RPC response a JSON object.

  - *Rationale*: If a RPC response is not a JSON object, then it is harder to avoid API breakage if
    new data in the response is needed.

Reply:
Interesting, I forgot about that. Thanks for reminding.

Reply:
It's a great guideline, glad it's written down!